### PR TITLE
Adgeneration: Update request to include device and app related info in headers and query params

### PR DESCRIPTION
**Pull request by banakemi**
## Type of change
- [x] Other

## Description of change
Adding ad request parameters.

- contact email of the adapter’s maintainer
  - Maintainer: ssp-ope@supership.jp
- [x] official adapter submission

### CODE REVIEW

**Comment by mansinahar** [comment]
@banakemi Can you please add JSON tests for the changes made in this PR?

**Comment by banakemi** [comment]
Hi @mansinahar 
Thanks for the review. Fixed it.

File: adapters/adgeneration/adgeneration.go [hunk 1]
```diff
@@ -55,8 +55,13 @@ func (adg *AdgenerationAdapter) MakeRequests(request *openrtb2.BidRequest, reqIn
 	headers := http.Header{}
 	headers.Add("Content-Type", "application/json;charset=utf-8")
 	headers.Add("Accept", "application/json")
-	if request.Device != nil && len(request.Device.UA) > 0 {
-		headers.Add("User-Agent", request.Device.UA)
+	if request.Device != nil {
+		if len(request.Device.UA) > 0 {
+			headers.Add("User-Agent", request.Device.UA)
+		}
```
Comment (mansinahar) [inline adapters/adgeneration/adgeneration.go:61]: @banakemi Can you please also add/modify existing JSON tests for the changes you've made in this file?

File: adapters/adgeneration/adgeneration.go [hunk 2]
```diff
@@ -112,12 +116,32 @@ func (adg *AdgenerationAdapter) getRawQuery(id string, request *openrtb2.BidRequ
 	if adSize != "" {
 		v.Set("sizes", adSize)
 	}
+	if request.Device != nil && request.Device.OS == "android" {
+		v.Set("sdktype", "1")
+	} else if request.Device != nil && request.Device.OS == "ios" {
+		v.Set("sdktype", "2")
+	} else {
+		v.Set("sdktype", "0")
+	}
```
Comment (AlexBVolcy) [inline adapters/adgeneration/adgeneration.go:125]: Just want to reiterate the need for json tests for these new conditionals!

File: adapters/adgeneration/adgeneration.go [hunk 3]
```diff
@@ -112,12 +116,32 @@ func (adg *AdgenerationAdapter) getRawQuery(id string, request *openrtb2.BidRequ
...
 	if request.Site != nil && request.Site.Page != "" {
 		v.Set("tp", request.Site.Page)
 	}
 	if request.Source != nil && request.Source.TID != "" {
 		v.Set("transactionid", request.Source.TID)
```
Comment (AlexBVolcy) [inline adapters/adgeneration/adgeneration.go:130]: This line isn't new code, but I checked the test coverage and it isn't covering this case, so we'll need a json test for this too!

File: adapters/adgeneration/adgeneration.go (hunk 1 above)
Comment (banakemi) [reply adapters/adgeneration/adgeneration.go:61]: @mansinahar 
Thank you. I fixed it.

File: adapters/adgeneration/adgeneration.go (hunk 3 above)
Comment (banakemi) [reply adapters/adgeneration/adgeneration.go:130]: @AlexBVolcy 
Thank you. I fixed it.

**Review by AlexBVolcy — APPROVED** [review]
Thank you for responding to feedback, the code coverage looks great!

**Review by mansinahar — APPROVED** [review]

**Pull request merged**